libs/api.lua

```diff
@@ -136,29 +200,32 @@ return function (prefix)
       return next(authors) and authors
     end,
     "^/search/(.*)$", function (query)
+      -- escape all special pattern characters except . and *
+      query = query:gsub("[%(%)%%%+%-%?%[%]%^%$]", function(c) return "%" .. c end)
```

Comment:
If we're going to escape most lua patterns, then we should probably support using star globs.

Your lua pattern-fu seems better than mine, but if it helps, I started implementing glob matches on domains in weblit at https://github.com/creationix/weblit/blob/master/libs/weblit-app.lua#L224-L239